Client/Frontend/Browser/Playlist/PlaylistSharedFolder.swift

```diff
@@ -0,0 +1,184 @@
+          // Items were added
+          continuation.resume(returning: folder)
+        }
+      }
+    }
+  }
+  
+  @MainActor
+  public static func saveToDiskStorage(memoryFolder: PlaylistFolder) async -> String {
+    await withCheckedContinuation({ continuation in
+      PlaylistFolder.saveInMemoryFolderToDisk(folder: memoryFolder) { folderId in
+        PlaylistItem.saveInMemoryItemsToDisk(items: Array(memoryFolder.playlistItems ?? []), folderUUID: folderId) {
+          continuation.resume(returning: folderId)
+        }
+      }
+    })
+  }
+  
+  private static func fetchMediaItemInfo(item: PlaylistSharedFolderModel) async -> [PlaylistInfo] {
+    @Sendable @MainActor
+    func fetchTask(item: PlaylistInfo) async -> PlaylistInfo {
+      await withCheckedContinuation { continuation in
+        var webLoader: PlaylistWebLoader?
+        webLoader = PlaylistWebLoader(
+          certStore: nil,
+          handler: { newItem in
+            if let newItem = newItem {
+              PlaylistManager.shared.getAssetDuration(item: newItem) { duration in
+                let item = PlaylistInfo(name: item.name,
+                                   src: newItem.src,
+                                   pageSrc: newItem.pageSrc,
+                                   pageTitle: item.pageTitle,
+                                   mimeType: newItem.mimeType,
+                                   duration: duration ?? newItem.duration,
+                                   detected: newItem.detected,
+                                   dateAdded: newItem.dateAdded,
+                                   tagId: item.tagId)
+                
+                // Destroy the web loader when the callback is complete.
+                webLoader?.removeFromSuperview()
+                webLoader = nil
+                continuation.resume(returning: item)
+              }
+            } else {
+              // Destroy the web loader when the callback is complete.
+              webLoader?.removeFromSuperview()
+              webLoader = nil
+              continuation.resume(returning: item)
+            }
+          }
+        ).then {
+          UIApplication.shared.keyWindow?.insertSubview($0, at: 0)
+        }
+
+        if let url = URL(string: item.pageSrc) {
+          webLoader?.load(url: url)
+        } else {
+          webLoader = nil
+        }
+      }
+    }
+
+    return await withTaskGroup(of: PlaylistInfo.self, returning: [PlaylistInfo].self) { group in
+      item.mediaItems.forEach { item in
+        group.addTask {
+          return await fetchTask(item: item)
+        }
+      }
+      
+      var result = [PlaylistInfo]()
+      for await value in group {
+        result.append(value)
+      }
+      return result
+    }
+  }
+  
+  private struct MediaItem: Codable {
+    public init() {
+      mediaItemId = UUID().uuidString
+      title = ""
+      url = NSURL() as URL
+    }
```

Comment:
This should be removed, given you have `let` properties so creating it with empty properties doesn't seem valid (technically an empty URL by using `NSURL` is not really valid either)

Reply:
Fixed, not needed at all.